### PR TITLE
Feat: WebPrompts - Dismissal Logic

### PR DESCRIPTION
**Pull request by rgomezp**
# Description
## 1 Line Summary
Contains changes related to the tracking of dismissal of web prompts.

## Details
We need to be able to track dismissal logic and follow the existing push notification re-prompt logic, that is, after 3, 7, and 30 days for the existing (push, category) and new (web prompts) slidedowns.

### Commits
* Revert "Fix Update Mode Bug" (wasn't a bug)
* Dismiss Model Implementation
* `DismissHelper` Update
* Update files to use new Dismiss function name change
* Nit: use different variable name
* Allow event listener to respond to multiple on CANCEL events

See individual commits for more info.

## 
This change is [Reviewable](https://reviewable.io/reviews/onesignal/onesignal-website-sdk/775)

### CODE REVIEW

File: src/models/Dismiss.ts [hunk 1]
```diff
@@ -0,0 +1,16 @@
+export enum DismissPrompt {
+  Push = 'push',
+  Web  = 'web' // (non-push)
```
Comment (jkasten2) [inline src/models/Dismiss.ts:3]: Web, is to confusing here. Maybe channel?

File: src/models/Dismiss.ts [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+}
+
+export enum DismissCountKey {
+PromptDismissCount = 'promptDismissCount', // legacy. applies to 'push' and 'category' slidedown types + native
```
Comment (jkasten2) [inline src/models/Dismiss.ts:7]: Missing space.

File: src/managers/slidedownManager/SlidedownManager.ts [hunk 3]
```diff
@@ -36,10 +36,10 @@ export class SlidedownManager {
     }
 
     private async checkIfSlidedownShouldBeShown(options: AutoPromptOptions): Promise<boolean> {
-        const wasDismissed = MainHelper.wasHttpsNativePromptDismissed();
         const permissionDenied = await OneSignal.privateGetNotificationPermission() === NotificationPermission.Denied;
         const isSubscribed = await OneSignal.privateIsPushNotificationsEnabled();
         const notOptedOut = await OneSignal.privateGetSubscription();
+        let wasDismissed;
```
Comment (jkasten2) [inline src/managers/slidedownManager/SlidedownManager.ts:42]: Missing type

File: src/models/Dismiss.ts (hunk 1 above)
Comment (rgomezp) [reply src/models/Dismiss.ts]: Decided on `NonPush` per discussion

**Pull request merged**